src/extractor/guidance/sliproad_handler.cpp

```diff
@@ -135,13 +135,13 @@ operator()(const NodeID /*nid*/, const EdgeID source_edge_id, Intersection inter
     if (intersection_node_id == next->node)
         return intersection;
 
-    std::unordered_set<NameID> target_road_names;
+    std::vector<NameID> target_road_name_ids;
+    target_road_name_ids.reserve(next->intersection.size());
```

Comment:
@MoKob I checked the sizes on Berlin, turns out it's really small.
We're fine with reserving here and then simply looping below.

Ticketed the `string_ref` idea in https://github.com/Project-OSRM/osrm-backend/issues/3265.